apps/emqx_resource/src/emqx_resource_metrics.erl

```diff
@@ -118,41 +120,58 @@ handle_telemetry_event(
     _Metadata = #{resource_id := ID},
     _HandlerConfig
 ) ->
-    case Event of
-        dropped_other ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped.other', Val);
-        dropped_expired ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped.expired', Val);
-        dropped_queue_full ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped.queue_full', Val);
-        dropped_resource_not_found ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped.resource_not_found', Val);
-        dropped_resource_stopped ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'dropped.resource_stopped', Val);
-        late_reply ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'late_reply', Val);
-        failed ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'failed', Val);
-        matched ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'matched', Val);
-        received ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'received', Val);
-        retried_failed ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'retried', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'failed', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'retried.failed', Val);
-        retried_success ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'retried', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'success', Val),
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'retried.success', Val);
-        success ->
-            emqx_metrics_worker:inc(?RES_METRICS, ID, 'success', Val);
-        _ ->
+    try
+        case Event of
```

Comment:
maybe turn this into a function with clauses for less indentation.

Reply:
:heavy_check_mark: